### PR TITLE
Use Hive version to determine if merge is supported

### PR DESCRIPTION
**Pull request by tovganesh**
Add a macro to get the Hive version to determine if the incremental merge is supported by warehouse.

The PR will check the Hive version for an incremental merge run and if the version is not 3, will fail.

### CODE REVIEW

File: dbt/include/hive/macros/adapters.sql [hunk 1]
```diff
@@ -258,4 +257,20 @@
...
+  {% set version_results = run_query('select version()') %}
+  {% if execute %}
+     {% set num_rows = (version_results | length) %}
+
+     {% if num_rows == 1 %}
+        {% set version_text = version_results[0][0] %}
+        {{ log("get_hive_version " ~ version_text) }}
+        {% do return(version_text.split(".")[0]) %}
+     {% else %}
+        return '2';  {# assume hive 2 by default #}
```
Comment (TapasSenapati) [inline dbt/include/hive/macros/adapters.sql:271]: is L270-272 redundant

Comment (tovganesh) [reply dbt/include/hive/macros/adapters.sql]: I don't think it is redundant, the macro containing query is typically parsed and then executed, so we need to handle the executed state separately (that is when the actual query results) are available as against parsing state. Ref: https://docs.getdbt.com/reference/dbt-jinja-functions/execute.
There was a typo though the in the return statement that I have fixed. 

Also attaching screenshots of Hive 2 vs 3 runs for incremental model with merge for reference.
image
image

File: dbt/include/hive/macros/adapters.sql [hunk 2]
```diff
@@ -258,4 +257,20 @@
...
 {% endmacro %}
-k
+
+{% macro get_hive_version() %}
+  {% set version_results = run_query('select version()') %}
+  {% if execute %}
+     {% set num_rows = (version_results | length) %}
+
+     {% if num_rows == 1 %}
+        {% set version_text = version_results[0][0] %}
```
Comment (sujitkp-blr) [inline dbt/include/hive/macros/adapters.sql:267]: can u give an sample of how version_results/version_text would look in case on hive 2 and hive 3

Comment (tovganesh) [reply dbt/include/hive/macros/adapters.sql:267]: Sample for Hive 2:
<pre>
0: jdbc:hive2://localhost:10000> select version();
+--------------------------------------------------+
|                       _c0                        |
+--------------------------------------------------+
| 2.3.2 r857a9fd8ad725a53bd95c1b2d6612f9b1155f44d  |
+--------------------------------------------------+
</pre>

Sample for Hive 3:
<pre>
+--------------------------------------------------+
|                       _c0                        |
+--------------------------------------------------+
| 3.1.3000.7.2.15.1-1 r386e2de0b8d5cb3f1803ec755243d43185505de6 |
+--------------------------------------------------+
</pre>

**Review by sujitkp-blr — APPROVED** [review]

**Pull request merged**